File: kotlin/src/commonMain/kotlin/lib/humanoid/paromat/Paromat.kt
package lib.humanoid.paromat

import lib.humanoid.paromat.interpreter.Interpreter
import lib.humanoid.paromat.lexer.Lexer
import lib.humanoid.paromat.parser.Parser

class Paromat {

    private val interpreter = Interpreter()

    public fun evaluate(expression: String): Number {
        return interpreter
            .visit(
                Parser(
                    Lexer(expression)
                ).parse()
            )
    }
}

File: python/paromat/paromat.py
from cmath import nan
from lexer import Lexer
from node_parser import Parser
from interpreter import Interpreter

def evaluate(expr: str):
    lex = Lexer(expr)
    tokens = lex.generateTokens()
    parser = Parser(tokens)
    parse_tree = parser.parse()
    if parse_tree:
        result = Interpreter().visit(parse_tree)
        return result.value
    else:
        return nan
